Updated highchart initializer to accept secondary argument, 'let' clean up

huiyuan opened the pull request:
Updated highchart initializer to accept secondary argument, removed extra 'let' to keep only one 'let' per scope.

poteto commented on addon/components/high-charts.js:
```diff
@@ -18,16 +18,17 @@ export default Ember.Component.extend({
-    }];
-
-    let defaults = { series: chartContent };
+    let defaultChartData = [{
```
I would revert `L24` to `L31` - https://github.com/airbnb/javascript#variables

Makes things harder to maintain.

poteto commented on addon/components/high-charts.js:
```diff
@@ -58,15 +59,16 @@ export default Ember.Component.extend({
   draw() {
-    let options = [ get(this, 'buildOptions') ];
-    let mode  = get(this, 'mode');
+    let completeChartOptions = [ get(this, 'buildOptions'), get(this, 'callback') ],
```
Same here, use one `let` per variable.

huiyuan replied:
I think both are fine and my main consideration here is to improve code readability and collaboration. Since you can declare variables anywhere in a function, it's better to keep them at one place so the developers who are not familiar with the code base won't make the trivial mistake to re-declare an existing variable. 

Besides, since var/let support multiple declarations, declaring one let per variable is a waste of code.

poteto replied:
I definitely appreciate the intent, I'm glad you're concerned about improving readability and collaboration. However, jshint will check for variables being redeclared, so that's a non-issue. 

Plus having multiple declarations makes it harder to maintain – now you have to remember to use a comma. To quote the airbnb style guide:

> Use one `var` declaration per variable. It's easier to add new variable declarations this way, and you never have to worry about swapping out a `;` for a `,` or introducing punctuation-only diffs.

The main thing is consistency. We don't currently do multiple declarations, so I'd like to keep it consistent and not do that.

cc @ahmadsoe

ahmadsoe replied:
Since this is ember addon, I prefer using  [Ember.js JavaScript Style Guide](https://github.com/emberjs/ember.js/blob/master/STYLEGUIDE.md).

@huiyuan so, please update your code to follow those rules before I can merge, especially about [variables](https://github.com/emberjs/ember.js/blob/master/STYLEGUIDE.md#variables).

huiyuan replied:
Good point, I'll update it.

poteto commented on addon/components/high-charts.js:
```diff
@@ -18,16 +18,17 @@ export default Ember.Component.extend({
+          data  : 0,
+          color : '#aaaaaa'
+        }],
+        chartOptions = Ember.$.extend(true, {}, get(this, 'theme'), get(this, 'chartOptions')),
```
One more thing: use [`Ember.merge`](http://emberjs.com/api/#method_merge)  instead of jQuery

huiyuan replied:
A deep merge is needed here. Ember.merge doesn't do that.

huiyuan commented:
@ahmadsoe , I reverted the variables.

huiyuan commented:
@ahmadsoe, can you merge please?

ahmadsoe commented:
@huiyuan it's conflicts. please re-update.

huiyuan commented:
@ahmadsoe , the conflicts are resolved.

ahmadsoe commented on index.js:
```diff
@@ -18,6 +20,10 @@ module.exports = {
   },
 
+  treeForVendor: function() {
+    return this.treeGenerator(path.join(__dirname, 'bower_components'));
```
why adding this? this will definitely break the addon.

huiyuan commented:
@ahmadsoe , I overlooked that one. I actually didn't add that back, it's probably from a bad merge. I just removed it and added back blueprints/.jshintrc. Sorry about that.

poteto commented:
Can you rebase this into a single commit please?

huiyuan commented:
I'll close this PR and open another with one single commit.
